synapse/handlers/relations.py

```diff
@@ -453,19 +405,43 @@ async def get_bundled_aggregations(
             latest_thread_event = thread.latest_event
             if latest_thread_event and latest_thread_event.event_id not in events_by_id:
                 events_by_id[latest_thread_event.event_id] = latest_thread_event
+                # The latest event in the thread must have a thread relation.
```

Comment:
I'm not following this. "must have" according to whom? Evidently it doesn't have a thread relation, otherwise we wouldn't have to do this?

Reply:
The latest event is an event which has a `rel_type` = `m.thread` to an event. (That's what we just searched for above.) The latest events are *new events* found, not part of the original input set, so we're expanding the information we calculated above.

Reply:
yup, much clearer, thanks.